### ISSUE TITLE
[Extension: Bug]: Frontmatter Syntax Highlighting Breaking

### ISSUE THREAD

**Issue by archiewood** [evidence-dev/evidence#2313]
### Describe the bug

When I have >1 line of YAML in my frontmatter, the syntax highlighting breaks

![CleanShot 2024-08-08 at 18 55 06@2x](https://github.com/user-attachments/assets/e2458869-4f0e-4b0b-af83-72b37ff47b59)

![CleanShot 2024-08-08 at 18 53 40@2x](https://github.com/user-attachments/assets/0567c90d-e869-4b2a-91e6-527b71f49cf7)

### Steps to Reproduce

```
---
title: Hello
variable: 5
---

Here is the rest of the markdown 

<BarChart
    data={query_name}
    x=column_name
    y=column_name
/>
```

### Severity

annoyance

### Additional Information, or Workarounds

~~I was able to resolve this by enabling the following VSCode extension
https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml~~

Upgrade VSCode to latest (>=1.92.1)

**Comment by archiewood** [comment]
this appears to be a VSCode bug (https://github.com/microsoft/vscode/issues/224862) with the default YAML grammar

The YAML extension install above is the recommended fix

**Comment by archiewood** [comment]
This is now fixed if you upgrade to latest VSCode (1.92.1)

https://github.com/microsoft/vscode/issues/224862#issuecomment-2277966697

**Comment by RedCMD** [comment]
caused by https://github.com/microsoft/vscode/pull/219833

you will need to update your `frontMatter` integration to something like https://github.com/microsoft/vscode-markdown-tm-grammar/pull/162
https://github.com/evidence-dev/evidence/blob/0d88330ff158bd83e58826c36c6eb29876f24e16/packages/extension/syntaxes/emd.tmLanguage.json#L506-L515

before VSCode re-adds the new YAML grammar again

related: https://github.com/microsoft/vscode/issues/224862
